### PR TITLE
Make rocksdb aware of jemalloc

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Signed-off-by: Yi Wu <yiwu@pingcap.com>

## What have you changed? (mandatory)
When jemalloc feature is enabled, also enable engine/jemalloc. This is to make rocksdb aware of existence of jemalloc. It doesn't affect which allocator to use. Mostly it only affect how rocksdb calculate block cache size: if jemalloc feature is on, rocksdb will use `malloc_usable_size()` (actual allocation size by jemalloc) to calculate size of block cache entries, instead of the data size of each entry, and it will lead to more accurate memory consumption calculation.

## What are the type of the changes? (mandatory)
Improvement

## How has this PR been tested? (mandatory)
CI

## Does this PR affect documentation (docs) or release note? (mandatory)
no

## Does this PR affect tidb-ansible update? (mandatory)
no

## Refer to a related PR or issue link (optional)
https://github.com/pingcap/rust-rocksdb/pull/319

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
Regarding CI failure, I think I should extract jemalloc header path from jemallocator crate and pass to rocksdb on rust-rocksdb side. Will fix.

**Comment by overvenus** [comment]
Should we cherry-pick to release-3.0?

**Comment by yiwu-arbug** [comment]
@overvenus I think no because it changes block cache size calculation anyway, so user may see cache performance difference and may need to adjust block cache size accordingly, which is not good for minor version update.

File: components/engine/Cargo.toml [hunk 1]
```diff
@@ -32,8 +33,9 @@ default-features = false
 features = ["nightly"]
 
 [dependencies.engine_rocksdb]
-git = "https://github.com/pingcap/rust-rocksdb.git"
+git = "https://github.com/yiwu-arbug/rust-rocksdb.git"
```
Comment (breeswish) [inline components/engine/Cargo.toml]: Should change back to pingcap?

Comment (yiwu-arbug) [reply components/engine/Cargo.toml]: Need an approval on https://github.com/pingcap/rust-rocksdb/pull/320

Comment (yiwu-arbug) [reply components/engine/Cargo.toml]: updated

**Comment by yiwu-arbug** [comment]
@breeswish mind give an approval again? Thanks.

**Comment by yiwu-arbug** [comment]
With this change build on Mac will get compile error. Will soon send a patch to not set the flag on Mac.

**Pull request merged**